src/app/screens/sendStx/steps/Step2SelectAmount.tsx

```diff
@@ -0,0 +1,233 @@
+      const newAmount = new BigNumber(stxBalance).minus(stxToMicrostacks(new BigNumber(fee ?? 0)));
+
+      if (newAmount.isGreaterThan(new BigNumber(0))) {
+        setAmount(newAmount.toString());
+      }
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [sendMax, fee]);
+
+  const { data: stxPendingTxData } = useStxPendingTxData();
+  const selectedNetwork = useNetworkSelector();
+
+  const [fees, setFees] = useState<FeeRates>({});
+
+  // Reactively construct a tx
+  useEffect(() => {
+    setIsLoading(true);
+
+    const createTx = async () => {
+      const rawUnsignedSendStxTx: StacksTransaction =
+        await generateUnsignedStxTokenTransferTransaction(
+          recipientAddress,
+          amount,
+          memo,
+          stxPendingTxData?.pendingTransactions ?? [],
+          stxPublicKey,
+          selectedNetwork,
+        );
+
+      // This function directly modifies unsignedSendStxTx lol
+      applyFeeMultiplier(rawUnsignedSendStxTx, feeMultipliers);
+      setUnsignedSendStxTx(buf2hex(rawUnsignedSendStxTx.serialize()));
+      setIsLoading(false);
+    };
+
+    createTx();
+  }, [
+    amount,
+    feeMultipliers,
+    memo,
+    recipientAddress,
+    selectedNetwork,
+    setIsLoading,
+    setUnsignedSendStxTx,
+    stxPendingTxData?.pendingTransactions,
+    stxPublicKey,
+  ]);
+
+  // Reactively estimate fees
+  useEffect(() => {
+    const fetchStxFees = async () => {
+      // const txRaw: string = await getRawTransaction(transaction.txid, network);
+
+      const unsignedTx: StacksTransaction = deserializeTransaction(unsignedSendStxTx);
+
+      const [low, medium, high] = await estimateTransaction(
+        unsignedTx.payload,
+        undefined,
+        selectedNetwork,
+      );
+      setFees({
+        low: Number(microstacksToStx(new BigNumber(low.fee)).toFixed(2)),
+        medium: Number(microstacksToStx(new BigNumber(medium.fee)).toFixed(2)),
+        high: Number(microstacksToStx(new BigNumber(high.fee)).toFixed(2)),
+      });
+      if (!fee)
+        setFeeRate(Number(microstacksToStx(new BigNumber(medium.fee)).toFixed(2)).toString());
+    };
+
+    fetchStxFees();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedNetwork, unsignedSendStxTx]);
+
+  return (
+    <Container>
+      <div>
+        {header}
+        <StxAmountSelector
+          amount={amount}
+          setAmount={setAmount}
+          sendMax={sendMax}
+          setSendMax={setSendMax}
+          disabled={!hasStx}
+        />
+        {hasStx && (
+          <FeeRateContainer>
+            <SelectFeeRate
+              fee={fee}
+              feeUnits="STX"
+              feeRate={feeRate}
+              setFeeRate={setFeeRate}
+              baseToFiat={stxToFiat}
+              fiatUnit={fiatCurrency}
+              getFeeForFeeRate={getFeeForFeeRate}
+              feeRates={fees}
+              feeRateLimits={{ min: 0.000001, max: feeMultipliers?.thresholdHighStacksFee }}
```

Comment:
is this min/max logic stated somewhere in requirements?

Reply:
No, it's not documented. It's the same logic used for the current send stx implementation.